polkadot/node/network/approval-distribution/src/lib.rs

```diff
@@ -24,9 +24,14 @@
 #![warn(missing_docs)]
 
 use self::metrics::Metrics;
-use futures::{channel::oneshot, select, FutureExt as _};
+use futures::{select, FutureExt as _};
 use itertools::Itertools;
 use net_protocol::peer_set::{ProtocolVersion, ValidationVersion};
+use polkadot_node_core_approval_voting::{
+	criteria::InvalidAssignment,
+	time::{Clock, ClockExt, SystemClock},
+	AssignmentCriteria, RealAssignmentCriteria, TICK_TOO_FAR_IN_FUTURE,
+};
```

Comment:
i feel a bit uneasy about approval-distribution depending on approval-voting crate
maybe extracting that part into polkadot_node_primitives::approval would make sense?

Reply:
Looking into it.

Reply:
Done, I extracted the assignment criteria interfaces and the time module from approval-voting in polkadot_node_primitives::approval, now approval-distribution does not depend at all on approval-voting.

Also, something to note here is that approval-distribution  and approval-voting, are meant to become a single subsystem after this refactoring series is completely finished, see https://github.com/paritytech/polkadot-sdk/pull/4849#issue-2364261568, so I was not to worried about the dependency.